libuuu/tar.h

```diff
@@ -0,0 +1,136 @@
+#include <stdint.h>
+#include <map>
+#include <string>
+#include <vector>
+#include "zlib.h"
+#include <memory>
+#include "buffer.h"
+#include "liberror.h"
+#include "libuuu.h"
+#include <stdio.h>
+#include <string.h>
+
+using namespace std;
+
+#define TAR_BLOCK_SIZE 512
+
+struct Tar_header
+{
+	char name[100];
+	char mode[8];
+	char owner_id[8];
+	char group_id[8];
+	char size[12];
+	char modi_time[12];
+	char checksum[8];
+	char type[1];
+	char linkname[100];
+	char ustar[6];
+	char version[2];
+	char uname[32];
+	char gname[32];
+	char major_num[8];
+	char minor_num[8];
+	char prefix[155];
+};
+
+
+
+class Tar_file_Info
+{
+public:
+	string filename;
+	uint64_t offset;
+	uint64_t size;
+
+};
+
+
+class Tar
+{
+	string filename;
+
+public:
+	map<string, Tar_file_Info> m_filemap;
+	int Open(string filename)
+	{
+		this->filename=filename;
+		ifstream stream(filename, ifstream::binary);
+		if(!stream)
```

Comment:
use get_file_buffer(filename) should be simpler than ifstream.  Previous zip implement is not good enough.